ext/liquid_c/block.c

```diff
@@ -56,6 +56,10 @@ static VALUE rb_block_parse(VALUE self, VALUE tokens, VALUE options)
             case TOKEN_RAW:
             {
                 VALUE str = rb_enc_str_new(token.str, token.length, utf8_encoding);
+
+                if(token.trim_whitespace)
+                   rb_funcall(str, rb_intern("lstrip!"), 0);
```

Comment:
I'm now thinking that it could be a good idea to make `read_while` respect the codepoint length (see other usages of `read_while` for what it does).